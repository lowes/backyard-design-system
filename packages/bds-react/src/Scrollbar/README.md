# @lowes-tech/bds-react/Scrollbar

## Interfaces

- [ScrollbarProps](interfaces/ScrollbarProps.md)

## References

### default

Renames and re-exports [Scrollbar](README.md#scrollbar)

## Type aliases

### ScrollbarRef

Ƭ **ScrollbarRef**: `HTMLDivElement`

## Variables

### Scrollbar

• **Scrollbar**: `BDSForwardRef`<[`ScrollbarProps`](interfaces/ScrollbarProps.md)\>

A styled scrollbar for use with the Backyard Design System.

TODO Implement a vertical scrollbar for the Y axis. Currently only provides a horizontal scrollbar for the X axis.

ex:
<Scrollbar>
    Bacon ipsum dolor amet t-bone cow flank tri-tip shankle tenderloin landjaeger rump alcatra capicola chicken
    pork chop. Filet mignon tenderloin drumstick burgdoggen swine shoulder picanha meatball ham venison chislic
    landjaeger andouille beef ribs jerky. Strip steak kevin sirloin cow picanha pork loin jowl chuck short loin
    shoulder sausage flank, brisket drumstick. Landjaeger ham pig meatball chuck ribeye. Kevin filet mignon
    pastrami turducken. Spare ribs pastrami chicken, meatball rump ball tip bacon kevin pork belly shoulder
    venison capicola cow pig ground round. Shank beef turducken, leberkas chuck tri-tip hamburger pork ham spare
    ribs kielbasa.
</Scrollbar>

note:
To override/change the scrollbar styling please target the following css classes:
  - .scrollbar-container
  - .scrollbar-track
  - .scrollbar-knob
To override/change the content styling please target the following css classes:
  - .content-container
