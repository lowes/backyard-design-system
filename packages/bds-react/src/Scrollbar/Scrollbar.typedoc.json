{
	"id": 0,
	"name": "@lowes-tech/bds-react/Scrollbar",
	"kind": 1,
	"kindString": "Project",
	"flags": {},
	"children": [
		{
			"id": 12,
			"name": "default",
			"kind": 16777216,
			"kindString": "Reference",
			"flags": {},
			"target": 1
		},
		{
			"id": 2,
			"name": "ScrollbarProps",
			"kind": 256,
			"kindString": "Interface",
			"flags": {},
			"children": [
				{
					"id": 10,
					"name": "as",
					"kind": 1024,
					"kindString": "Property",
					"flags": {
						"isOptional": true
					},
					"type": {
						"type": "reference",
						"typeArguments": [
							{
								"type": "intrinsic",
								"name": "any"
							}
						],
						"name": "ElementType"
					},
					"inheritedFrom": {
						"type": "reference",
						"name": "BackyardBaseProps.as"
					}
				},
				{
					"id": 3,
					"name": "className",
					"kind": 1024,
					"kindString": "Property",
					"flags": {
						"isOptional": true
					},
					"comment": {
						"shortText": "Classname for the ScrollbarWrapper"
					},
					"type": {
						"type": "intrinsic",
						"name": "string"
					},
					"overwrites": {
						"type": "reference",
						"name": "BackyardBaseProps.className"
					}
				},
				{
					"id": 5,
					"name": "id",
					"kind": 1024,
					"kindString": "Property",
					"flags": {
						"isOptional": true
					},
					"comment": {
						"shortText": "Id for the content the ScrollbarWrapper is wrapping"
					},
					"type": {
						"type": "intrinsic",
						"name": "string"
					},
					"overwrites": {
						"type": "reference",
						"name": "BackyardBaseProps.id"
					}
				},
				{
					"id": 4,
					"name": "wrapperProps",
					"kind": 1024,
					"kindString": "Property",
					"flags": {
						"isOptional": true
					},
					"comment": {
						"shortText": "Additional props for the ScrollbarWrapper"
					},
					"type": {
						"type": "reference",
						"typeArguments": [
							{
								"type": "reference",
								"name": "HTMLDivElement"
							},
							{
								"type": "literal",
								"value": ""
							}
						],
						"name": "BackyardBaseProps"
					}
				},
				{
					"id": 8,
					"name": "onResize",
					"kind": 2048,
					"kindString": "Method",
					"flags": {
						"isOptional": true
					},
					"signatures": [
						{
							"id": 9,
							"name": "onResize",
							"kind": 4096,
							"kindString": "Call signature",
							"flags": {},
							"type": {
								"type": "intrinsic",
								"name": "void"
							}
						}
					]
				},
				{
					"id": 6,
					"name": "onScroll",
					"kind": 2048,
					"kindString": "Method",
					"flags": {
						"isOptional": true
					},
					"signatures": [
						{
							"id": 7,
							"name": "onScroll",
							"kind": 4096,
							"kindString": "Call signature",
							"flags": {},
							"comment": {
								"shortText": "Used to pass additional functionality to the onScroll event"
							},
							"type": {
								"type": "intrinsic",
								"name": "void"
							},
							"overwrites": {
								"type": "reference",
								"name": "BackyardBaseProps.onScroll"
							}
						}
					],
					"overwrites": {
						"type": "reference",
						"name": "BackyardBaseProps.onScroll"
					}
				}
			],
			"groups": [
				{
					"title": "Properties",
					"kind": 1024,
					"children": [
						10,
						3,
						5,
						4
					]
				},
				{
					"title": "Methods",
					"kind": 2048,
					"children": [
						8,
						6
					]
				}
			],
			"extendedTypes": [
				{
					"type": "reference",
					"typeArguments": [
						{
							"type": "reference",
							"id": 11,
							"name": "ScrollbarRef"
						}
					],
					"name": "BackyardBaseProps"
				}
			]
		},
		{
			"id": 11,
			"name": "ScrollbarRef",
			"kind": 4194304,
			"kindString": "Type alias",
			"flags": {},
			"type": {
				"type": "reference",
				"name": "HTMLDivElement"
			}
		},
		{
			"id": 1,
			"name": "Scrollbar",
			"kind": 32,
			"kindString": "Variable",
			"flags": {},
			"comment": {
				"shortText": "A styled scrollbar for use with the Backyard Design System.",
				"text": "TODO Implement a vertical scrollbar for the Y axis. Currently only provides a horizontal scrollbar for the X axis.\n\nex:\n<Scrollbar>\n    Bacon ipsum dolor amet t-bone cow flank tri-tip shankle tenderloin landjaeger rump alcatra capicola chicken\n    pork chop. Filet mignon tenderloin drumstick burgdoggen swine shoulder picanha meatball ham venison chislic\n    landjaeger andouille beef ribs jerky. Strip steak kevin sirloin cow picanha pork loin jowl chuck short loin\n    shoulder sausage flank, brisket drumstick. Landjaeger ham pig meatball chuck ribeye. Kevin filet mignon\n    pastrami turducken. Spare ribs pastrami chicken, meatball rump ball tip bacon kevin pork belly shoulder\n    venison capicola cow pig ground round. Shank beef turducken, leberkas chuck tri-tip hamburger pork ham spare\n    ribs kielbasa.\n</Scrollbar>\n\nnote:\nTo override/change the scrollbar styling please target the following css classes:\n  - .scrollbar-container\n  - .scrollbar-track\n  - .scrollbar-knob\nTo override/change the content styling please target the following css classes:\n  - .content-container\n"
			},
			"type": {
				"type": "reference",
				"typeArguments": [
					{
						"type": "reference",
						"id": 2,
						"name": "ScrollbarProps"
					}
				],
				"name": "BDSForwardRef"
			},
			"defaultValue": "..."
		}
	],
	"groups": [
		{
			"title": "References",
			"kind": 16777216,
			"children": [
				12
			]
		},
		{
			"title": "Interfaces",
			"kind": 256,
			"children": [
				2
			]
		},
		{
			"title": "Type aliases",
			"kind": 4194304,
			"children": [
				11
			]
		},
		{
			"title": "Variables",
			"kind": 32,
			"children": [
				1
			]
		}
	]
}