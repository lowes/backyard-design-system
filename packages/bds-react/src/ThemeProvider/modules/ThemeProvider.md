# Namespace: ThemeProvider

## Variables

### bdsName

â€¢ **bdsName**: `string`
