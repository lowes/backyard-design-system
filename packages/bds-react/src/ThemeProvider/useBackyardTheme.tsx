import * as React from 'react'

import { ThemeContext } from './ThemeProvider'
import type { BackyardTheme } from './ThemeProvider'

const validateBackyardTheme = (theme: BackyardTheme, name: BackyardTheme['name']) => {
    if (!theme?.name) {
        // eslint-disable-next-line no-console
        throw Error(
            `Backyard: A Backyard component ${
                name ? `(\`${name}\`)` : ''
            } is missing a Backyard Theme.\n` +
                'This component uses Backyard tokens and a Backyard Theme is required.\n' +
                `Please wrap ${
                    name ? `\`${name}\` or` : ''
                } your entire application in a Backyard \`ThemeProvider\`.\n` +
                'For a getting started guide, see \n' +
                '`https://backyard.lowes.com/GetStarted/Develop#create-theme-provider`',
        )
    }

    return true
}

const useBackyardTheme = (props: UseBackyardThemeProps = {}) => {
    const { validate = false, name }: UseBackyardThemeProps = props

    const theme = React.useContext<BackyardTheme>(ThemeContext)

    if (validate && process.env.NODE_ENV !== 'production') {
        React.useEffect(() => {
            validateBackyardTheme(theme, name)
        }, [theme])
    }

    return theme
}

interface UseBackyardThemeProps {
    /**
     * Whether or not to validate component
     */
    validate?: boolean
    /**
     * Name of the component to validate
     */
    name?: string
}

export { useBackyardTheme, validateBackyardTheme }

export default useBackyardTheme
