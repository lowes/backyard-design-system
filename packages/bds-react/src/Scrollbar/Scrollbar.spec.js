import * as React from 'react'
import { render } from '../../test-utils'
import { ThemeProvider } from '../ThemeProvider'
import Scrollbar from './Scrollbar'

describe('Scrollbar Snapshots', () => {
    test('Scrollbar default snapshot', () => {
        const { asFragment } = render(
            <Scrollbar id="test">
                Bacon ipsum dolor amet t-bone cow flank tri-tip shankle tenderloin landjaeger rump
                alcatra capicola chicken pork chop. Filet mignon tenderloin drumstick burgdoggen
                swine shoulder picanha meatball ham venison chislic landjaeger andouille beef ribs
                jerky. Strip steak kevin sirloin cow picanha pork loin jowl chuck short loin
                shoulder sausage flank, brisket drumstick. Landjaeger ham pig meatball chuck ribeye.
                Kevin filet mignon pastrami turducken. Spare ribs pastrami chicken, meatball rump
                ball tip bacon kevin pork belly shoulder venison capicola cow pig ground round.
                Shank beef turducken, leberkas chuck tri-tip hamburger pork ham spare ribs kielbasa.
            </Scrollbar>,
        )

        expect(asFragment()).toMatchSnapshot()
    })

    test('wrapperProps prop snapshot', () => {
        const { asFragment } = render(
            <Scrollbar id="test" wrapperProps={{ style: { backgroundColor: 'green' } }}>
                Bacon ipsum dolor amet t-bone cow flank tri-tip shankle tenderloin landjaeger rump
                alcatra capicola chicken pork chop. Filet mignon tenderloin drumstick burgdoggen
                swine shoulder picanha meatball ham venison chislic landjaeger andouille beef ribs
                jerky. Strip steak kevin sirloin cow picanha pork loin jowl chuck short loin
                shoulder sausage flank, brisket drumstick. Landjaeger ham pig meatball chuck ribeye.
                Kevin filet mignon pastrami turducken. Spare ribs pastrami chicken, meatball rump
                ball tip bacon kevin pork belly shoulder venison capicola cow pig ground round.
                Shank beef turducken, leberkas chuck tri-tip hamburger pork ham spare ribs kielbasa.
            </Scrollbar>,
        )

        expect(asFragment()).toMatchSnapshot()
    })

    test('className prop snapshot', () => {
        const { asFragment } = render(
            <Scrollbar id="test" className="test-class">
                Bacon ipsum dolor amet t-bone cow flank tri-tip shankle tenderloin landjaeger rump
                alcatra capicola chicken pork chop. Filet mignon tenderloin drumstick burgdoggen
                swine shoulder picanha meatball ham venison chislic landjaeger andouille beef ribs
                jerky. Strip steak kevin sirloin cow picanha pork loin jowl chuck short loin
                shoulder sausage flank, brisket drumstick. Landjaeger ham pig meatball chuck ribeye.
                Kevin filet mignon pastrami turducken. Spare ribs pastrami chicken, meatball rump
                ball tip bacon kevin pork belly shoulder venison capicola cow pig ground round.
                Shank beef turducken, leberkas chuck tri-tip hamburger pork ham spare ribs kielbasa.
            </Scrollbar>,
        )

        expect(asFragment()).toMatchSnapshot()
    })

    describe('themes', () => {
        const themes = ['light', 'dark']

        themes.forEach((theme) => {
            test(`Scrollbar ${theme} theme snapshot`, () => {
                const { asFragment } = render(
                    <ThemeProvider theme={theme}>
                        <React.Fragment>
                            <Scrollbar id="test">
                                Bacon ipsum dolor amet t-bone cow flank tri-tip shankle tenderloin
                                landjaeger rump alcatra capicola chicken pork chop. Filet mignon
                                tenderloin drumstick burgdoggen swine shoulder picanha meatball ham
                                venison chislic landjaeger andouille beef ribs jerky. Strip steak
                                kevin sirloin cow picanha pork loin jowl chuck short loin shoulder
                                sausage flank, brisket drumstick. Landjaeger ham pig meatball chuck
                                ribeye. Kevin filet mignon pastrami turducken. Spare ribs pastrami
                                chicken, meatball rump ball tip bacon kevin pork belly shoulder
                                venison capicola cow pig ground round. Shank beef turducken,
                                leberkas chuck tri-tip hamburger pork ham spare ribs kielbasa.
                            </Scrollbar>
                        </React.Fragment>
                    </ThemeProvider>,
                )

                expect(asFragment()).toMatchSnapshot()
            })
        })
    })

    describe('extra props', () => {
        test('Scrollbar DOM props (id, className) snapshot', () => {
            const { asFragment } = render(
                <Scrollbar id="id" className="classname">
                    hi Bacon ipsum dolor amet t-bone cow flank tri-tip shankle tenderloin landjaeger
                    rump alcatra capicola chicken pork chop. Filet mignon tenderloin drumstick
                    burgdoggen swine shoulder picanha meatball ham venison chislic landjaeger
                    andouille beef ribs jerky. Strip steak kevin sirloin cow picanha pork loin jowl
                    chuck short loin shoulder sausage flank, brisket drumstick. Landjaeger ham pig
                    meatball chuck ribeye. Kevin filet mignon pastrami turducken. Spare ribs
                    pastrami chicken, meatball rump ball tip bacon kevin pork belly shoulder venison
                    capicola cow pig ground round. Shank beef turducken, leberkas chuck tri-tip
                    hamburger pork ham spare ribs kielbasa.
                </Scrollbar>,
            )

            expect(asFragment()).toMatchSnapshot()
        })
    })
})
