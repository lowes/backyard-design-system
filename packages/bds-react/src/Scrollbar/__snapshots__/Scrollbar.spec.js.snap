// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scrollbar Snapshots Scrollbar default snapshot 1`] = `
<DocumentFragment>
  .c0 {
  height: auto;
  width: auto;
}

.c0 .content-container {
  overflow-x: scroll;
  -ms-overflow-style: none;
  -webkit-scrollbar-width: none;
  -moz-scrollbar-width: none;
  -ms-scrollbar-width: none;
  scrollbar-width: none;
}

.c0 .content-container::-webkit-scrollbar {
  display: none;
}

.c0 .content-container .scrollbar-content {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c1 {
  width: 100%;
  padding-top: var(--bds-sizes-size-4);
  padding-bottom: var(--bds-sizes-size-4);
}

.c1:hover {
  cursor: pointer;
}

.c1.hide-scrollbar {
  display: none;
}

.c3 {
  position: absolute;
  left: 0;
  width: 10%;
  height: var(--bds-sizes-size-4);
  background-color: var(--bds-color-action-interactive);
  border-radius: var(--bds-border-radius-circle);
  -webkit-transition: top 0.08s;
  -moz-transition: top 0.08s;
  -ms-transition: top 0.08s;
  -o-transition: top 0.08s;
  -webkit-transition: top 0.08s;
  transition: top 0.08s;
}

.c3:hover {
  background-color: var(--bds-color-action-interactive-hover);
}

.c3:hover,
.c3:focus,
.c3:active {
  margin-top: -0.0625rem;
  height: var(--bds-sizes-size-6);
}

.c2 {
  position: relative;
  background: var(--bds-color-surface-subdued);
  height: 4px;
  -webkit-order: 10;
  -ms-flex-order: 10;
  order: 10;
  width: 100%;
  border-radius: var(--bds-border-radius-circle);
}

<div
    class="c0"
  >
    <div
      class="content-container"
      id="test"
    >
      <div
        class="scrollbar-content"
      >
        Bacon ipsum dolor amet t-bone cow flank tri-tip shankle tenderloin landjaeger rump alcatra capicola chicken pork chop. Filet mignon tenderloin drumstick burgdoggen swine shoulder picanha meatball ham venison chislic landjaeger andouille beef ribs jerky. Strip steak kevin sirloin cow picanha pork loin jowl chuck short loin shoulder sausage flank, brisket drumstick. Landjaeger ham pig meatball chuck ribeye. Kevin filet mignon pastrami turducken. Spare ribs pastrami chicken, meatball rump ball tip bacon kevin pork belly shoulder venison capicola cow pig ground round. Shank beef turducken, leberkas chuck tri-tip hamburger pork ham spare ribs kielbasa.
      </div>
    </div>
    <div
      class="c1 scrollbar-container hide-scrollbar"
    >
      <div
        class="c2 scrollbar-track"
      >
        <div
          aria-controls="test"
          class="c3 scrollbar-knob"
          role="scrollbar"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Scrollbar Snapshots className prop snapshot 1`] = `
<DocumentFragment>
  .c0 {
  height: auto;
  width: auto;
}

.c0 .content-container {
  overflow-x: scroll;
  -ms-overflow-style: none;
  -webkit-scrollbar-width: none;
  -moz-scrollbar-width: none;
  -ms-scrollbar-width: none;
  scrollbar-width: none;
}

.c0 .content-container::-webkit-scrollbar {
  display: none;
}

.c0 .content-container .scrollbar-content {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c1 {
  width: 100%;
  padding-top: var(--bds-sizes-size-4);
  padding-bottom: var(--bds-sizes-size-4);
}

.c1:hover {
  cursor: pointer;
}

.c1.hide-scrollbar {
  display: none;
}

.c3 {
  position: absolute;
  left: 0;
  width: 10%;
  height: var(--bds-sizes-size-4);
  background-color: var(--bds-color-action-interactive);
  border-radius: var(--bds-border-radius-circle);
  -webkit-transition: top 0.08s;
  -moz-transition: top 0.08s;
  -ms-transition: top 0.08s;
  -o-transition: top 0.08s;
  -webkit-transition: top 0.08s;
  transition: top 0.08s;
}

.c3:hover {
  background-color: var(--bds-color-action-interactive-hover);
}

.c3:hover,
.c3:focus,
.c3:active {
  margin-top: -0.0625rem;
  height: var(--bds-sizes-size-6);
}

.c2 {
  position: relative;
  background: var(--bds-color-surface-subdued);
  height: 4px;
  -webkit-order: 10;
  -ms-flex-order: 10;
  order: 10;
  width: 100%;
  border-radius: var(--bds-border-radius-circle);
}

<div
    class="c0 test-class"
  >
    <div
      class="content-container"
      id="test"
    >
      <div
        class="scrollbar-content"
      >
        Bacon ipsum dolor amet t-bone cow flank tri-tip shankle tenderloin landjaeger rump alcatra capicola chicken pork chop. Filet mignon tenderloin drumstick burgdoggen swine shoulder picanha meatball ham venison chislic landjaeger andouille beef ribs jerky. Strip steak kevin sirloin cow picanha pork loin jowl chuck short loin shoulder sausage flank, brisket drumstick. Landjaeger ham pig meatball chuck ribeye. Kevin filet mignon pastrami turducken. Spare ribs pastrami chicken, meatball rump ball tip bacon kevin pork belly shoulder venison capicola cow pig ground round. Shank beef turducken, leberkas chuck tri-tip hamburger pork ham spare ribs kielbasa.
      </div>
    </div>
    <div
      class="c1 scrollbar-container hide-scrollbar"
    >
      <div
        class="c2 scrollbar-track"
      >
        <div
          aria-controls="test"
          class="c3 scrollbar-knob"
          role="scrollbar"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Scrollbar Snapshots extra props Scrollbar DOM props (id, className) snapshot 1`] = `
<DocumentFragment>
  .c0 {
  height: auto;
  width: auto;
}

.c0 .content-container {
  overflow-x: scroll;
  -ms-overflow-style: none;
  -webkit-scrollbar-width: none;
  -moz-scrollbar-width: none;
  -ms-scrollbar-width: none;
  scrollbar-width: none;
}

.c0 .content-container::-webkit-scrollbar {
  display: none;
}

.c0 .content-container .scrollbar-content {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c1 {
  width: 100%;
  padding-top: var(--bds-sizes-size-4);
  padding-bottom: var(--bds-sizes-size-4);
}

.c1:hover {
  cursor: pointer;
}

.c1.hide-scrollbar {
  display: none;
}

.c3 {
  position: absolute;
  left: 0;
  width: 10%;
  height: var(--bds-sizes-size-4);
  background-color: var(--bds-color-action-interactive);
  border-radius: var(--bds-border-radius-circle);
  -webkit-transition: top 0.08s;
  -moz-transition: top 0.08s;
  -ms-transition: top 0.08s;
  -o-transition: top 0.08s;
  -webkit-transition: top 0.08s;
  transition: top 0.08s;
}

.c3:hover {
  background-color: var(--bds-color-action-interactive-hover);
}

.c3:hover,
.c3:focus,
.c3:active {
  margin-top: -0.0625rem;
  height: var(--bds-sizes-size-6);
}

.c2 {
  position: relative;
  background: var(--bds-color-surface-subdued);
  height: 4px;
  -webkit-order: 10;
  -ms-flex-order: 10;
  order: 10;
  width: 100%;
  border-radius: var(--bds-border-radius-circle);
}

<div
    class="c0 classname"
  >
    <div
      class="content-container"
      id="id"
    >
      <div
        class="scrollbar-content"
      >
        hi Bacon ipsum dolor amet t-bone cow flank tri-tip shankle tenderloin landjaeger rump alcatra capicola chicken pork chop. Filet mignon tenderloin drumstick burgdoggen swine shoulder picanha meatball ham venison chislic landjaeger andouille beef ribs jerky. Strip steak kevin sirloin cow picanha pork loin jowl chuck short loin shoulder sausage flank, brisket drumstick. Landjaeger ham pig meatball chuck ribeye. Kevin filet mignon pastrami turducken. Spare ribs pastrami chicken, meatball rump ball tip bacon kevin pork belly shoulder venison capicola cow pig ground round. Shank beef turducken, leberkas chuck tri-tip hamburger pork ham spare ribs kielbasa.
      </div>
    </div>
    <div
      class="c1 scrollbar-container hide-scrollbar"
    >
      <div
        class="c2 scrollbar-track"
      >
        <div
          aria-controls="id"
          class="c3 scrollbar-knob"
          role="scrollbar"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Scrollbar Snapshots themes Scrollbar dark theme snapshot 1`] = `
<DocumentFragment>
  .c0 {
  height: auto;
  width: auto;
}

.c0 .content-container {
  overflow-x: scroll;
  -ms-overflow-style: none;
  -webkit-scrollbar-width: none;
  -moz-scrollbar-width: none;
  -ms-scrollbar-width: none;
  scrollbar-width: none;
}

.c0 .content-container::-webkit-scrollbar {
  display: none;
}

.c0 .content-container .scrollbar-content {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c1 {
  width: 100%;
  padding-top: var(--bds-sizes-size-4);
  padding-bottom: var(--bds-sizes-size-4);
}

.c1:hover {
  cursor: pointer;
}

.c1.hide-scrollbar {
  display: none;
}

.c3 {
  position: absolute;
  left: 0;
  width: 10%;
  height: var(--bds-sizes-size-4);
  background-color: var(--bds-color-action-interactive);
  border-radius: var(--bds-border-radius-circle);
  -webkit-transition: top 0.08s;
  -moz-transition: top 0.08s;
  -ms-transition: top 0.08s;
  -o-transition: top 0.08s;
  -webkit-transition: top 0.08s;
  transition: top 0.08s;
}

.c3:hover {
  background-color: var(--bds-color-action-interactive-hover);
}

.c3:hover,
.c3:focus,
.c3:active {
  margin-top: -0.0625rem;
  height: var(--bds-sizes-size-6);
}

.c2 {
  position: relative;
  background: var(--bds-color-surface-subdued);
  height: 4px;
  -webkit-order: 10;
  -ms-flex-order: 10;
  order: 10;
  width: 100%;
  border-radius: var(--bds-border-radius-circle);
}

<div
    class="c0"
  >
    <div
      class="content-container"
      id="test"
    >
      <div
        class="scrollbar-content"
      >
        Bacon ipsum dolor amet t-bone cow flank tri-tip shankle tenderloin landjaeger rump alcatra capicola chicken pork chop. Filet mignon tenderloin drumstick burgdoggen swine shoulder picanha meatball ham venison chislic landjaeger andouille beef ribs jerky. Strip steak kevin sirloin cow picanha pork loin jowl chuck short loin shoulder sausage flank, brisket drumstick. Landjaeger ham pig meatball chuck ribeye. Kevin filet mignon pastrami turducken. Spare ribs pastrami chicken, meatball rump ball tip bacon kevin pork belly shoulder venison capicola cow pig ground round. Shank beef turducken, leberkas chuck tri-tip hamburger pork ham spare ribs kielbasa.
      </div>
    </div>
    <div
      class="c1 scrollbar-container hide-scrollbar"
    >
      <div
        class="c2 scrollbar-track"
      >
        <div
          aria-controls="test"
          class="c3 scrollbar-knob"
          role="scrollbar"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Scrollbar Snapshots themes Scrollbar light theme snapshot 1`] = `
<DocumentFragment>
  .c0 {
  height: auto;
  width: auto;
}

.c0 .content-container {
  overflow-x: scroll;
  -ms-overflow-style: none;
  -webkit-scrollbar-width: none;
  -moz-scrollbar-width: none;
  -ms-scrollbar-width: none;
  scrollbar-width: none;
}

.c0 .content-container::-webkit-scrollbar {
  display: none;
}

.c0 .content-container .scrollbar-content {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c1 {
  width: 100%;
  padding-top: var(--bds-sizes-size-4);
  padding-bottom: var(--bds-sizes-size-4);
}

.c1:hover {
  cursor: pointer;
}

.c1.hide-scrollbar {
  display: none;
}

.c3 {
  position: absolute;
  left: 0;
  width: 10%;
  height: var(--bds-sizes-size-4);
  background-color: var(--bds-color-action-interactive);
  border-radius: var(--bds-border-radius-circle);
  -webkit-transition: top 0.08s;
  -moz-transition: top 0.08s;
  -ms-transition: top 0.08s;
  -o-transition: top 0.08s;
  -webkit-transition: top 0.08s;
  transition: top 0.08s;
}

.c3:hover {
  background-color: var(--bds-color-action-interactive-hover);
}

.c3:hover,
.c3:focus,
.c3:active {
  margin-top: -0.0625rem;
  height: var(--bds-sizes-size-6);
}

.c2 {
  position: relative;
  background: var(--bds-color-surface-subdued);
  height: 4px;
  -webkit-order: 10;
  -ms-flex-order: 10;
  order: 10;
  width: 100%;
  border-radius: var(--bds-border-radius-circle);
}

<div
    class="c0"
  >
    <div
      class="content-container"
      id="test"
    >
      <div
        class="scrollbar-content"
      >
        Bacon ipsum dolor amet t-bone cow flank tri-tip shankle tenderloin landjaeger rump alcatra capicola chicken pork chop. Filet mignon tenderloin drumstick burgdoggen swine shoulder picanha meatball ham venison chislic landjaeger andouille beef ribs jerky. Strip steak kevin sirloin cow picanha pork loin jowl chuck short loin shoulder sausage flank, brisket drumstick. Landjaeger ham pig meatball chuck ribeye. Kevin filet mignon pastrami turducken. Spare ribs pastrami chicken, meatball rump ball tip bacon kevin pork belly shoulder venison capicola cow pig ground round. Shank beef turducken, leberkas chuck tri-tip hamburger pork ham spare ribs kielbasa.
      </div>
    </div>
    <div
      class="c1 scrollbar-container hide-scrollbar"
    >
      <div
        class="c2 scrollbar-track"
      >
        <div
          aria-controls="test"
          class="c3 scrollbar-knob"
          role="scrollbar"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Scrollbar Snapshots wrapperProps prop snapshot 1`] = `
<DocumentFragment>
  .c0 {
  height: auto;
  width: auto;
}

.c0 .content-container {
  overflow-x: scroll;
  -ms-overflow-style: none;
  -webkit-scrollbar-width: none;
  -moz-scrollbar-width: none;
  -ms-scrollbar-width: none;
  scrollbar-width: none;
}

.c0 .content-container::-webkit-scrollbar {
  display: none;
}

.c0 .content-container .scrollbar-content {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c1 {
  width: 100%;
  padding-top: var(--bds-sizes-size-4);
  padding-bottom: var(--bds-sizes-size-4);
}

.c1:hover {
  cursor: pointer;
}

.c1.hide-scrollbar {
  display: none;
}

.c3 {
  position: absolute;
  left: 0;
  width: 10%;
  height: var(--bds-sizes-size-4);
  background-color: var(--bds-color-action-interactive);
  border-radius: var(--bds-border-radius-circle);
  -webkit-transition: top 0.08s;
  -moz-transition: top 0.08s;
  -ms-transition: top 0.08s;
  -o-transition: top 0.08s;
  -webkit-transition: top 0.08s;
  transition: top 0.08s;
}

.c3:hover {
  background-color: var(--bds-color-action-interactive-hover);
}

.c3:hover,
.c3:focus,
.c3:active {
  margin-top: -0.0625rem;
  height: var(--bds-sizes-size-6);
}

.c2 {
  position: relative;
  background: var(--bds-color-surface-subdued);
  height: 4px;
  -webkit-order: 10;
  -ms-flex-order: 10;
  order: 10;
  width: 100%;
  border-radius: var(--bds-border-radius-circle);
}

<div
    class="c0"
    style="background-color: green;"
  >
    <div
      class="content-container"
      id="test"
    >
      <div
        class="scrollbar-content"
      >
        Bacon ipsum dolor amet t-bone cow flank tri-tip shankle tenderloin landjaeger rump alcatra capicola chicken pork chop. Filet mignon tenderloin drumstick burgdoggen swine shoulder picanha meatball ham venison chislic landjaeger andouille beef ribs jerky. Strip steak kevin sirloin cow picanha pork loin jowl chuck short loin shoulder sausage flank, brisket drumstick. Landjaeger ham pig meatball chuck ribeye. Kevin filet mignon pastrami turducken. Spare ribs pastrami chicken, meatball rump ball tip bacon kevin pork belly shoulder venison capicola cow pig ground round. Shank beef turducken, leberkas chuck tri-tip hamburger pork ham spare ribs kielbasa.
      </div>
    </div>
    <div
      class="c1 scrollbar-container hide-scrollbar"
    >
      <div
        class="c2 scrollbar-track"
      >
        <div
          aria-controls="test"
          class="c3 scrollbar-knob"
          role="scrollbar"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
